app/uk/gov/hmrc/plasticpackagingtax/registration/connectors/GrsConnector.scala

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.plasticpackagingtax.registration.connectors
+
+import com.kenshoo.play.metrics.Metrics
+import play.api.http.Status.CREATED
+import play.api.libs.json.{Reads, Writes}
+import uk.gov.hmrc.http.{HeaderCarrier, HttpClient, HttpResponse, InternalServerException}
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+abstract class GrsConnector[
```

Comment:
Since this is `abstract` should it even have an `@Inject` annotation? Personally I'd probably just have all its dependencies defined as abstract methods on the class instead, then an implementation can decide how it provides those dependencies, e.g.

```scala
abstract class A {
  protected def foo: String
}

class B @Inject() (override protected val foo: String) extends A 

class C extends A {
  override protected def foo: String = "C"
}
```

Reply:
`@Inject` annotation removed. Dependencies provided in ctor for consistency with other classes in the project.